pkg/nodeconfig/nodeconfig.go

```diff
@@ -346,25 +346,26 @@ func (nc *nodeConfig) createFilesFromIgnition() (map[string]string, error) {
 		return nil, err
 	}
 
-	filesToTransfer := map[string]struct{}{
-		ignition.KubeletCACertPath: {},
-	}
-	if _, ok := kubeletArgs[ignition.CloudConfigOption]; ok {
-		filesToTransfer[ignition.CloudConfigPath] = struct{}{}
-	}
 	filePathsToContents := make(map[string]string)
-	// For each new file in the ignition file check if is a file we are interested in and, if so, decode its contents
-	for _, ignFile := range ign.GetFiles() {
-		if _, ok := filesToTransfer[ignFile.Node.Path]; ok {
+	// process kubelet-ca
+	filePathsToContents[windows.K8sDir+"\\"+KubeletClientCAFilename] = string(ign.GetKubeletCAData())
+	// process cloud config, if enabled
+	if _, ok := kubeletArgs[ignition.CloudConfigOption]; ok {
+		for _, ignFile := range ign.GetFiles() {
+			if ignFile.Node.Path != ignition.CloudConfigPath {
+				continue
+			}
 			if ignFile.Contents.Source == nil {
-				return nil, fmt.Errorf("could not process %s: File is empty", ignFile.Node.Path)
+				return nil, fmt.Errorf("could not process %s: File is empty", ignition.CloudConfigPath)
 			}
 			contents, err := dataurl.DecodeString(*ignFile.Contents.Source)
 			if err != nil {
-				return nil, fmt.Errorf("could not decode %s: %w", ignFile.Node.Path, err)
+				return nil, fmt.Errorf("could not decode %s: %w", ignition.CloudConfigPath, err)
 			}
-			fileName := filepath.Base(ignFile.Node.Path)
+			fileName := filepath.Base(ignition.CloudConfigPath)
 			filePathsToContents[windows.K8sDir+"\\"+fileName] = string(contents.Data)
+
+			break
```

Comment:
This seems like an unnecessary refactor in a bug fix.

Reply:
@sebsoto  [Addressed](https://github.com/openshift/windows-machine-config-operator/compare/49047ec9fe7ee9ffdb78cbfae8d37835435eb3b3..e588e5c73f1d74403a52332e27b677bfe58de4e3)